### ISSUE TITLE
slider drag behavior when mouse is outside of slider bounding box

### ISSUE THREAD

**Issue by lumonix** [vurtun/nuklear#636]
In nk_slider_float (around line 22304 in my version):

`    in = (state == NK_WIDGET_ROM || layout->flags & NK_WINDOW_ROM) ? 0 : &ctx->input;`

The first test `NK_WIDGET_ROM` prevents the slider from functioning when you press down on the slider, then drag and move the mouse outside of the BB of the slider.

This does not seem like the desired behavior of a slider.

Disabling the first check seems to give me the behavior I want, but I am unsure what the side effect of this change may be.

The slider itself already seems to test if the mouse down press originated in the sliders BB and do the correct thing if it wasn't.

Suggestions?

**Comment by vurtun** [comment]
Hey @lumonix,
thanks for the issue. I checked and for slider it is possible to remove `state == NK_WIDGET_ROM`. 

The reason why this check was added is to prevent dragging for widgets that are only half visible. For most clickable widgets preventing widget to be interactive if only half visible is important but for dragging widgets like sliders it does not make sense.

Reason why sliders had that check was that I only later added dragging behavior to sliders and did not remove the condition. I pushed a new commit removing the check.
